test/trace/basic_server.js

```diff
@@ -163,3 +163,17 @@ test('basic tracing test', function (assert) {
         }, 10);
     });
 });
+
+test('modifying traceSample', function (assert){
+    var channel = new TChannel();
+    var subservice = channel.makeSubChannel({
+        serviceName: 'subservice'
+    });
+    assert.equal(channel.traceSample, 0.01);
+    assert.equal(subservice.traceSample, 0.01);
+
+    subservice.setTraceSample(0.99);
+    assert.equal(channel.traceSample, 0.99);
+    assert.equal(subservice.traceSample, 0.99);
```

Comment:
Alternately, we can consult this.topChannel.traceSample when deciding whether to trace each request, instead of this.traceSample, and only set traceSample on the topChannel. We should fail an assertion if you call setTraceSample on any channel except the topChannel so that we avoid action at a distance confusion.